torchvision/ops/roi_align.py

```diff
@@ -232,7 +250,9 @@ def roi_align(
     if not isinstance(rois, torch.Tensor):
         rois = convert_boxes_to_roi_format(rois)
     if not torch.jit.is_scripting():
-        if not _has_ops() or (torch.are_deterministic_algorithms_enabled() and (input.is_cuda or input.is_mps)):
+        if (
+            not _has_ops() or (torch.are_deterministic_algorithms_enabled() and (input.is_cuda or input.is_mps))
```

Comment:
Should we just remove the mps part here since you mentioned MPS doesn't even work with torch.compile?

```suggestion
            not _has_ops() or (torch.are_deterministic_algorithms_enabled() and input.is_cuda)
```

Reply:
I opted to keep it around, because it was explicitly added by @qqaatw, but I don't really mind either way

Reply:
Sorry for the late reply! I'm ok with either way that is best for the development. From the mentioned issue it seems only relevant to CUDA, is MPS similarly memory hungry with deterministic algorithm?